src/backend/common/CommandBuffer.cpp

```diff
@@ -597,14 +598,25 @@ namespace backend {
                             return false;
                         }
 
-                        // TODO(kainino@chromium.org): We should be able to optimize this by tracking which textures are in use as attachments.
-                        auto& subpassInfo = currentRenderPass->GetSubpassInfo(currentSubpass);
-                        for (auto attachmentSlot : subpassInfo.colorAttachments) {
-                            // XXX: this loop is running too long; how do we make it work correctly?
-                            auto* tv = currentFramebuffer->GetTextureView(attachmentSlot);
-                            if (tv && tv->GetTexture() == texture) {
-                                device->HandleError("Can't transition a texture while it's used as a color attachment");
-                                return false;
+                        // TODO(kainino@chromium.org): We should be able to
```

Comment:
I'm thinking that we could have scratch buffers in textures and buffers, that can be used during validation to store custom data such as this. Validation looks good like this for now.